core/modules/user/user.module

```diff
@@ -805,6 +812,13 @@ function user_account_form(&$form, &$form_state) {
     '#description' => t('A valid email address. All emails from the system will be sent to this address. The email address is not made public and will only be used if you wish to receive a new password or wish to receive certain news or notifications by email.'),
     '#required' => TRUE,
     '#default_value' => (!$register ? $account->mail : ''),
+    '#attributes' => array(
+      // Add attributes to field to prevent spell-jacking and unwanted automatic
+      // changes caused by the browser.
+      'autocapitalize' => 'none',
+      'autocorrect' => 'off',
+      'spellcheck' => 'false',
+    ),
```

Comment:
This block simply adds the desired attributes to both name and password for this specific high-level form.